AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -639,62 +646,93 @@ - (BOOL)isLayerBacked
   return _flags.layerBacked;
 }
 
-#pragma mark - Layout measurement calculation
+#pragma mark - Layout measurement and sizing
+
+- (ASSize)size
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  return _size;
+}
+
+- (void)setSize:(ASSize)size
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  if (ASSizeEqualToSize(_size, size) == NO) {
+    _size = size;
+    [self invalidateCalculatedLayout];
```

Comment:
Question: Should the invalidate call happen while the lock is acquired? `__instanceLock__` is recursive so this won't deadlock but I'm just wondering if the layout needs to be invalidated while this lock is acquired.